pkg/reconciler/route/route.go

```diff
@@ -270,7 +270,11 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 		return err
 	}
 
-	r.Status.PropagateIngressStatus(*ingress.GetStatus())
+	if ingress.GetObjectMeta().GetGeneration() != ingress.GetStatus().ObservedGeneration {
```

Comment:
This feels like a different change.

Reply:
See the description. This fixes a bug that was discovered because of the removal of the 404 logic.
I can split into 2 PR if you want but I put them together because there are linked:
* you need the bug fix to have the 404 retry logic removal to pass E2E tests
* the bug fix is not validated without the 404 retry logic change

Reply:
Ok, I moved it to https://github.com/knative/serving/pull/5057